### PR TITLE
Refactor the spec documentation with better structure:

### PR DESCRIPTION
**Pull request by resouer**
1. Clarify the intention that workload type is the NAME of workload definition.
2. Clarify ContainerizedWorkload as schema of Server workload type.
3. Create placeholder for Task workload type.
4. Fix several bugs
5. Remove SPEC_WORKING_DRAFT.md.
6. Rename SPEC_LATEST_STABLE.md to SPEC.md
7. Move specification components as part of SPEC.md, not readme.

### CODE REVIEW

File: 4.workload_definitions.md [hunk 1]
```diff
@@ -25,6 +23,28 @@ Here are the attributes that provide top-level information about the workload de
 
 The metadata section describes the workload definition. See [*metadata*](2.overview_and_terminology.md#Metadata).
 
+#### `Name`
+
+The `name` of the workload definition indicates _the type of this workload_.
```
Comment (hongchaodeng) [inline 4.workload_definitions.md:28]: This seems to be a major change to the spec.
I'm wondering what's the purpose to have such change?

Comment (resouer) [reply 4.workload_definitions.md:28]: See: https://github.com/crossplane/oam-kubernetes-runtime/issues/195

In Vela, multiple workload types will reuse same `definitionRef`.

**Review by hongchaodeng — APPROVED** [review]
lgtm

File: 4.workload_definitions.md [hunk 2]
```diff
@@ -25,6 +23,28 @@ Here are the attributes that provide top-level information about the workload de
...
+#### `Labels`
+
+Infrastructure operators could add below reserved `labels` to workload definition to indicates _the distinguishing characteristics of this workload type_.
+
+|Label|Type|Explain|
+|-|-|-|
+|`workload.oam.dev/replicable`|boolean|Whether they are replicable. If not, no replication or scaling traits may be assigned.|
+|`workload.oam.dev/daemonized`|boolean|Whether they are daemonized. For daemon types, if the workload exits, this is considered a fault, and the system must fix it. For non-daemonized types, exit is considered a success if no error is reported.|
+|`workload.oam.dev/exposed`|boolean|Whether they are exposed, i.e. have a service endpoint with a stable name for network traffic. Workload types that have a service endpoint need a virtual IP address (VIP) with a DNS name to represent the component as a whole, addressable within their network scope and can be assigned traffic routing traits.|
+|`workload.oam.dev/podspecable`|boolean|Whether this workload can be addressed by Kubernetes `PodSpec`. If yes, the implementation could manipulate the workload by leveraging `PodSpec` structure, instead of being agnostic of the workload's schematic. |
```
Comment (wonderflow) [inline 4.workload_definitions.md:46]: just for clarify, we haven't implemented this label mechanism yet.

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**